moveit_ros/moveit_servo/src/utilities.cpp

```diff
@@ -144,4 +144,44 @@ double velocityScalingFactorForSingularity(const moveit::core::JointModelGroup*
   return velocity_scale;
 }
 
+bool applyJointUpdate(rclcpp::Clock& clock, const double publish_period, const Eigen::ArrayXd& delta_theta,
+                      const sensor_msgs::msg::JointState& previous_joint_state,
+                      sensor_msgs::msg::JointState& next_joint_state,
+                      std::shared_ptr<online_signal_smoothing::SmoothingBaseClass>& smoother)
+{
+  // All the sizes must match
+  if (next_joint_state.position.size() != static_cast<std::size_t>(delta_theta.size()) ||
+      next_joint_state.velocity.size() != next_joint_state.position.size())
+  {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+    RCLCPP_ERROR_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD,
+                                 "Lengths of output and increments do not match.");
+#pragma GCC diagnostic pop
```

Comment:
This GCC stuff is new to me. What is this?

Reply:
This supresses the warning
```
 use of old-style cast to ‘rcutils_duration_value_t’ {aka ‘long int’} [-Wold-style-cast]
 RCLCPP_ERROR_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD
```
However a proper fix would be to do `static_cast`
```
constexpr auto ROS_LOG_THROTTLE_PERIOD = static_cast<rcutils_duration_value_t>(std::chrono::milliseconds(3000).count());
```

Reply:
best go for the proper fix. There are other compilers besides GCC  :+1:

Reply:
The fix has been applied , removed the older warning suppressions as well.

Reply:
This suppression is present in other files as well.
Might need to raise an issue for this.

Reply:
Unfortunately casting it does not seem to fix the issue.
[This](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/issues/955) issue seems to be deeper.

Reply:
ok, feel free to ignore my comment

Reply:
Have changed the warnings to global within `utilities`  rather than per call to keep things a bit cleaner.